### PR TITLE
[CELEBORN-601] Consolidate configsWithAlternatives with `ConfigBuilder.withAlternative`

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
 Consolidate configsWithAlternatives with `ConfigBuilder.withAlternative`
Current code if we use  below two function
```
  def get(key: String): String = {
    getOption(key).getOrElse(throw new NoSuchElementException(key))
  }


  def get(key: String, defaultValue: String): String = {
    getOption(key).getOrElse(defaultValue)
  }
```

They are not consist.

### CODE REVIEW

**Comment by pan3793** [comment]
what are they inconsistent before?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -995,6 +995,15 @@ object CelebornConf extends Logging {
     }.toMap
   }
 
+  def addDeprecatedConfig(entry: ConfigEntry[_], alt: (String, String => String)): Unit = {
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:998]: this function is dangerous to expose to the public scope. Currently, it was invoked by `register` and `unregister`, both methods are guarded by `confEntriesUpdateLock`, so it's actually no thread-safe issue. But ideally, `configsWithAlternatives` should have its own guard.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -979,7 +979,7 @@ object CelebornConf extends Logging {
    * The alternates are used in the order defined in this map. If deprecated configs are
    * present in the user's configuration, a warning is logged.
    */
-  private val configsWithAlternatives = Map[String, Seq[AlternateConfig]](
+  private val configsWithAlternatives = scala.collection.mutable.Map[String, Seq[AlternateConfig]](
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:982]: nit: `mutable.Map`

File: common/src/test/scala/org/apache/celeborn/common/CelebornConfSuite.scala [hunk 3]
```diff
@@ -177,4 +177,11 @@ class CelebornConfSuite extends CelebornFunSuite {
     assert(conf.networkConnectTimeout.duration.toMillis == 2000L)
     assert(conf.networkIoConnectTimeoutMs("data") == 2000L)
   }
+
+  test("CELEBORN-601: Consolidate configsWithAlternatives with `ConfigBuilder.withAlternative`") {
+    val conf = new CelebornConf()
+      .set("rss.network.timeout", "300s")
```
Comment (pan3793) [inline common/src/test/scala/org/apache/celeborn/common/CelebornConfSuite.scala:183]: `rss.` alias is going to be removed, maybe we can create a dedicated one for testing

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:982]: Done

File: common/src/test/scala/org/apache/celeborn/common/CelebornConfSuite.scala (hunk 3 above)
Comment (AngersZhuuuu) [reply common/src/test/scala/org/apache/celeborn/common/CelebornConfSuite.scala:183]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: How about the current? add `private` to it. and it won't be called in multiple thread.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -3308,4 +3323,13 @@ object CelebornConf extends Logging {
       .doc("the heartbeat interval between worker and client")
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("60s")
+
+  val TEST_ALTERNATIVE: OptionalConfigEntry[String] =
+    buildConf("celeborn.testAlternative.value")
+      .withAlternative("celeborn.testAlternative.replaceKey")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3329]: `celeborn.test.` is a namespace.

```suggestion
    buildConf("celeborn.test.alternative.key")
      .withAlternative("celeborn.test.alternative.deprecatedKey")
```

**Review by pan3793 — APPROVED** [review]
LGTM, only minor naming suggestion

**Pull request merged**